test/blackbox-tests/test-cases/melange/missing-melc.t

```diff
@@ -0,0 +1,86 @@
+Test cases when melc is not available
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.6)
+  > (using melange 0.1)
+  > EOF
+
+  $ cat > main.ml <<EOF
+  > let () =
+  >   print_endline "hello"
+  > EOF
+
+Set up some fake environment without melc
+
+  $ mkdir _path
+  $ ln -s $(command -v dune) _path/
+  $ ln -s $(command -v ocamlc) _path/
+
+For melange.emit stanzas, an error is shown
+
+  $ cat > dune <<EOF
+  > (melange.emit
+  >  (target output)
+  >  (module_system commonjs))
+  > EOF
+
+  $ (unset INSIDE_DUNE; PATH=_path dune runtest --always-show-command-line --root .)
+  Error: A melange.emit stanza was found, but the melange compiler melc is not
+  available. Either install it by running [opam install melange], or remove the
+  melange.emit stanzas from the project.
+  [1]
+
+For libraries, if no melange.emit stanza is found, build does not fail
```

Comment:
The build does not fail and this is the correct behavior. We don't need melange to build and run the executable.